lib/player/player.go

```diff
@@ -29,9 +29,12 @@ import (
 	"github.com/gravitational/trace"
 	"github.com/jonboulle/clockwork"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/maps"
 
 	"github.com/gravitational/teleport"
 	"github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/player/db"
```

Comment:
I would prefer to do this in a way that `lib/player` doesn't depend on any protocol-specific packages.

What if we just checked if the `events.AuditEvent` also implements something like `interface { Print() []byte }`?

Reply:
> What if we just checked if the events.AuditEvent also implements something like interface { Print() []byte }?

That wouldn't be enough to present the information the way we're currently doing. The "translator" needs context from other events to avoid presenting data with no correlation. This is only used in prepared statement scenarios, where we don't print messages on prepare/bind but store their information for later presentation on the execute event.

Alternatively, we could move this to the events package (or other related package) and receive it as a player configuration. We could make it not depend on anything protocol-specific but receive an events modifier.

For example:

```
type EventsModifier interface {
    // Define if the modifier should be used on the session recording.
    Use(startSessionEvt AuditEvent) bool
    // Process receives the session recording event and modifies it as necessary.
    Process(evt AuditEvent) (modfiedEvent AuditEvent, modified bool)
}

// We would provide a list of modifiers to the player when constructing it.
// Alternatively, it could hold a list of modifiers constructors.
player.New(player.Config{
    Modifiers: []events.EventsModifier{postgresmodifier.New()},
})
```

@zmb3 What do you think?

Reply:
Or maybe move the translator to a custom Streamer?

Reply:
> Or maybe move the translator to a custom Streamer?

Yeah, that might be an option too. Are you thinking of solving multiple translators on it (For example, `StreamerWithTranslators`) or having a streamer per protocol/translator (for example, `PostgresStreamer`)?

Reply:
i was thinking per protocol